raft/raft.go

```diff
@@ -242,6 +246,7 @@ func (r *raft) bcastHeartbeat() {
 		}
 		r.sendHeartbeat(i)
 		r.prs[i].resume()
+		r.prs[i].ins.reset()
```

Comment:
Maybe it should reset on receiving an MsgHeartbeatResp instead of sending an MsgHeartbeat. 

I wonder if it would be better to remove one request from inflights instead of all of them. This would guarantee that you have a minimum of one MsgApp per heartbeat interval in replicate mode, just like in probe mode. Once you have an MsgAppResp its index will clear out the rest of the inflights.